dataland-frontend/src/components/resources/frameworkDataSearch/p2p/P2pPanel.vue

```diff
@@ -0,0 +1,332 @@
+        } else {
+          this.p2pDataAndMetaInfo = (
+            await p2pDataControllerApi.getAllCompanyP2pData(assertDefined(this.companyId))
+          ).data;
+        }
+        this.convertP2pDataToFrontendFormat();
+        this.waitingForData = false;
+      } catch (error) {
+        console.error(error);
+      }
+    },
+
+    /**
+     * Creates kpi data objects to pass them to the data table.
+     * @param kpiKey The field name of a kpi
+     * @param kpiValue The corresponding value to the kpiKey
+     * @param subcategory The sub category to which the kpi belongs
+     * @param category category to which the kpi belongs to
+     * @param dataIdOfP2pDataset The value of the date kpi of an LkSG dataset
+     */
+    createKpiDataObjects(
+      kpiKey: string,
+      kpiValue: KpiValue,
+      subcategory: Subcategory,
+      category: Category,
+      dataIdOfP2pDataset: string
+    ): void {
+      const kpiField = assertDefined(subcategory.fields.find((field) => field.name === kpiKey));
+      const kpiData = {
+        categoryKey: category.name == "general" ? `_${category.name}` : category.name,
+        categoryLabel: category.label ? category.label : category.name,
+        subcategoryKey: subcategory.name == "general" ? `_${subcategory.name}` : subcategory.name,
+        subcategoryLabel: subcategory.label ? subcategory.label : subcategory.name,
+        kpiKey: kpiKey,
+        kpiLabel: kpiField?.label ? kpiField.label : kpiKey,
+        kpiDescription: kpiField?.description ? kpiField.description : "",
+        kpiFormFieldComponent: kpiField?.component ?? "",
+        content: { [dataIdOfP2pDataset]: reformatValueForDisplay(kpiField, kpiValue) },
+      } as KpiDataObject;
+      if (this.mapOfKpiKeysToDataObjects.has(kpiKey)) {
+        Object.assign(kpiData.content, this.mapOfKpiKeysToDataObjects.get(kpiKey)?.content);
+      }
+      this.mapOfKpiKeysToDataObjects.set(kpiKey, kpiData);
+      this.resultKpiData = kpiData;
+    },
+    /**
+     * Retrieves and converts the stored array of P2P datasets in order to make it displayable in the frontend.
+     */
+    convertP2pDataToFrontendFormat(): void {
+      if (this.p2pDataAndMetaInfo.length) {
+        this.p2pDataAndMetaInfo.forEach((oneP2pDataset: DataAndMetaInformationPathwaysToParisData) => {
+          const dataIdOfP2pDataset = oneP2pDataset.metaInfo?.dataId ?? "";
+          const reportingPeriodOfP2pDataset = oneP2pDataset.metaInfo?.reportingPeriod ?? "";
+          this.listOfDataSetReportingPeriods.push({
+            dataId: dataIdOfP2pDataset,
+            reportingPeriod: reportingPeriodOfP2pDataset,
+          });
+          for (const [categoryKey, categoryObject] of Object.entries(oneP2pDataset.data) as [string, object] | null) {
+            if (categoryObject == null) continue;
+            const listOfDataObjects: Array<KpiDataObject> = [];
```

Comment:
Why even initialize this here. We could get it as a return value from `iterateThroughSubcategories`.

Reply:
Let's have a short call for this to talk about